precompiles/dispatch-lockdrop/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+    <Runtime as Config>::AccountId: From<[u8; 32]>,
+    DispatchValidator:
+        DispatchValidateT<<Runtime as Config>::AccountId, <Runtime as Config>::RuntimeCall>,
+    DecodeLimit: Get<u32>,
+{
+    #[precompile::public("dispatch_lockdrop_call(bytes,bytes32,bytes)")]
+    fn dispatch_lockdrop_call(
+        handle: &mut impl PrecompileHandle,
+        call: UnboundedBytes,
+        account_id: H256,
+        signature: BoundedBytes<ECDSASignatureBytes>,
+    ) -> EvmResult<bool> {
+        log::trace!(
+            target: LOG_TARGET,
+            "raw arguments: call: {:?}, account_id: {:?}, signature: {:?}",
+            call,
+            account_id,
+            signature
+        );
+
+        let target_gas = handle.gas_limit();
+        let caller: H160 = handle.context().caller.into();
+        let input: Vec<u8> = call.into();
+        let signature_bytes: Vec<u8> = signature.into();
+        let account_id = AccountId32::new(account_id.into()).into();
+
+        // 1. Decode the call
+        let call =
+            match Runtime::RuntimeCall::decode_with_depth_limit(DecodeLimit::get(), &mut &*input) {
+                Ok(c) => c,
+                Err(_) => {
+                    log::trace!(target: LOG_TARGET, "Error: could not decode call");
+                    return Ok(false);
+                }
+            };
+
+        // 2. Check that gas limit is not exceeded
+        let info = call.get_dispatch_info();
+        if let Some(gas) = target_gas {
+            let valid_weight = info.weight.ref_time()
+                <= Runtime::GasWeightMapping::gas_to_weight(gas, false).ref_time();
+            if !valid_weight {
+                log::trace!(target: LOG_TARGET, "Error: gas limit exceeded");
+                return Ok(false);
+            }
+        }
+
+        // 3. Recover the ECDSA Public key from the signature
+        let signature_opt = match Self::parse_signature(&signature_bytes) {
+            Some(s) => s,
+            None => {
+                log::trace!(target: LOG_TARGET, "Error: could not parse signature");
+                return Ok(false);
+            }
+        };
+        let payload_hash = Self::build_signing_payload(&account_id);
+        let pubkey = match <pallet_unified_accounts::Pallet<Runtime>>::recover_pubkey(
+            signature_opt.as_ref(),
+            payload_hash,
+        ) {
+            Some(k) => k,
+            None => {
+                log::trace!(
+                    target: LOG_TARGET,
+                    "Error: could not recover pubkey from signature"
+                );
+                return Ok(false);
+            }
+        };
+
+        // 4. Ensure that the caller matches the recovered EVM address from the signature
+        if caller != Self::get_evm_address_from_pubkey(&pubkey) {
+            log::trace!(
+                target: LOG_TARGET,
+                "Error: caller does not match calculated EVM address"
+            );
+            return Ok(false);
```

Comment:
How come you don't revert here?
It seems more appropriate since the dispatch failed.

Same question for other occurrences.